Lib/netrc.py

```diff
@@ -81,13 +82,15 @@ def _parse(self, file, fp, default_netrc):
         lexer = _netrclex(fp)
         while 1:
             # Look for a machine, default, or macdef top-level keyword
-            saved_lineno = lexer.lineno
-            toplevel = tt = lexer.get_token()
+            tt = lexer.get_token()
             if not tt:
                 break
             elif tt[0] == '#':
-                if lexer.lineno == saved_lineno and len(tt) == 1:
+                # For top level tokens, we skip line if the # is followed
+                # by a space / newline. Otherwise, we only skip the token.
+                if tt == '#' and not lexer.dontskip:
```

Comment:
I'm not sure if I understand why the entirety of `t` is being compared to a `'#'`. Is this semantically “the entire token consists of just `#`”?
This could be
```suggestion
                if len(tt) == 1 and not lexer.dontskip:
```
but then, why does it matter whether it's `# thing` vs `#thing`? Typically, comment parsers just disregard whatever's after the hash and don't interpret that in any way…